src/sundials/sundials_profiler.c

```diff
@@ -496,22 +514,44 @@ int sunCompareTimes(const void* l, const void* r)
   double left_max;
   double right_max;
 
-  const SUNHashMapKeyValue left = *((SUNHashMapKeyValue*) l);
-  const SUNHashMapKeyValue right = *((SUNHashMapKeyValue*) r);
+  const SUNHashMapKeyValue left  = *((SUNHashMapKeyValue*)l);
+  const SUNHashMapKeyValue right = *((SUNHashMapKeyValue*)r);
+
+  if (left == NULL && right == NULL) return (0);
+  if (left == NULL) return (1);
+  if (right == NULL) return (-1);
 
-  if (left == NULL && right == NULL)
-    return(0);
-  if (left == NULL)
-    return(1);
-  if (right == NULL)
-    return(-1);
+  left_max  = ((sunTimerStruct*)left->value)->maximum;
+  right_max = ((sunTimerStruct*)right->value)->maximum;
+
+  if (left_max < right_max) return (1);
+  if (left_max > right_max) return (-1);
+  return (0);
+}
 
-  left_max = ((sunTimerStruct*) left->value)->maximum;
-  right_max = ((sunTimerStruct*) right->value)->maximum;
+int sunclock_gettime_monotonic(sunTimespec* ts)
+{
+#if (defined(WIN32) || defined(_WIN32)) && !defined(SUNDIALS_HAVE_POSIX_TIMERS)
+  static LARGE_INTEGER ticks_per_sec;
+  LARGE_INTEGER ticks;
 
-  if (left_max < right_max)
-    return(1);
-  if (left_max > right_max)
-    return(-1);
-  return(0);
+  if (!ticks_per_sec.QuadPart)
+  {
+    QueryPerformanceFrequency(&ticks_per_sec);
+    if (!ticks_per_sec.QuadPart)
+    {
+      return -1;
+    }
+  }
+
+  QueryPerformanceCounter(&ticks);
+
+  ts->tv_sec  = (long)(ticks.QuadPart / ticks_per_sec.QuadPart);
+  ts->tv_nsec = (long)(((ticks.QuadPart % ticks_per_sec.QuadPart) * 1000000000) /
```

Comment:
The Windows documentation says "1 microsecond or better" but I have not found a way to get the exact resolution.

Reply:
I think this is still right, since `ticks.QuadPart % ticks_per_sec.QuadPart` would give a remainder in seconds, multiplying it by 1,000,000,000 then yields the remainder in nanoseconds.